src/components/Autocomplete/components/types.ts

```diff
@@ -1,9 +1,4 @@
-import * as PropTypes from 'prop-types';
-import {ValidationMap} from 'react';
-
-export const contextTypes: ValidationMap<any> = {
-  selectedOptionId: PropTypes.string,
-  comboBoxId: PropTypes.string,
-  subscribe: PropTypes.func,
-  unsubscribe: PropTypes.func,
-};
+export interface ComboBoxContext {
```

Comment:
What do you think to moving this interface into the Context subcomponent so it can live with the Provider / Consumer?

Is there every any case where we'd need this but not the Provider or Consumer?

Reply:
Depends, the two cases I see using this outside of `Context.tsx` is for typing context in the component that creates it and passes it to the provider and if we ever want to create a hook for using the context (we probably won't).

Reply:
In the first case you'd be able to do `import {ComboBoxContext, Provider} from 'X';` so you'd only need one import instead of 2. If we wanted a hook then we'd need to access the result of `React.createContext` so having the type in the same file would save you an import there too.

I reckon stick it in the same file as where you create the context.

Reply:
Resolving this for now, as we can move the typing into a ComponentName/context.ts file in a follow up PR